### PR TITLE
Add subscribes for different paths.

### PR DESCRIPTION
**Pull request by platinumthinker**
I rework this library for more powerfull usage from multiple appllication (different subscribtion),
and add simply filtring monitoring files:
```
FilterFun = fun(Filename) -> filename:extension(".erl") == true end,
erlfsmon:subscribe("/home/", FilterFun, [modified, renamed])

```

### CODE REVIEW

**Comment by proger** [comment]
Thanks for the patch!

Why not filter paths on the subscriber side? Unless you're expecting to receive a huge amount of filesystem events per second :)

File: src/erlfsmon_server.erl [hunk 1]
```diff
@@ -3,56 +3,95 @@
...
-            port=Port,
-            path=Path,
-            backend=Backend
-        }}.
+init([Backend]) ->
+    {ok, #state{backend = Backend}}.
+
+handle_call({subscribe, Path, FileFilter, KnownEvents}, {Pid, _},
+            #state{backend = Backend, subscribers = Subs} = State) ->
+    Port = Backend:start_port(Path, Path),
```
Comment (proger) [inline src/erlfsmon_server.erl:42]: If you really insist on running multiple backends for disjoint paths, take a look at https://github.com/synrc/fs, it basically `start_link`s the same supervisor for each path.

**Comment by proger** [comment]
I've added [start_link_subscribe](https://github.com/proger/erlfsmon/commit/dd0f7e5b9f4de9c124362ff1f3c4907f72995609) API that uses a supervisor+gen_server per path (i find it slightly cleaner than implementation in fs), let me know what you think.

**Comment by platinumthinker** [comment]
Thanks for review my code. I'm more suited library fs, as you said

**Comment by platinumthinker** [comment]
I see API for fs: i can't subscibe several path..
If you don't want extend api, i for you lib or synrc/fs

**Comment by proger** [comment]
Have you tried the `start_link_subscribe` API yet?

```
~/src/erlfsmon% ERL_LIBS=deps erl -pa ebin
Erlang/OTP 19 [erts-8.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Eshell V8.1  (abort with ^G)
1> os:getenv("HOME").                                                  
"/Users/vladki"
2> erlfsmon:start_link_subscribe(os:getenv("HOME"), fswatch, home_events).
ok
3> erlfsmon:start_link_subscribe(".", fswatch, here_events).              
ok
4> flush().
Shell got {<0.61.0>,
           {erlfsmon_server,file_event},
           {"/Users/vladki/Library/Application Support/Google/Chrome/Default/Cookies-journal",
            ['PlatformSpecific','Updated','IsFile']}}
Shell got {<0.65.0>,
           {erlfsmon_server,file_event},
           {"/Users/vladki/src/erlfsmon/lol",['PlatformSpecific','IsFile']}}
ok
```

**Comment by platinumthinker** [comment]
Ok

**Pull request closed without merging**